ee/clickhouse/queries/funnels/funnel_event_query.py

```diff
@@ -44,22 +53,40 @@ def get_query(self, entities=None, entity_name="events", skip_entity_filter=Fals
             for column_name in self._column_optimizer.event_columns_to_query
         )
 
-        _fields.extend(
-            f"groups_{group_index}.group_properties_{group_index} as group_properties_{group_index}"
-            for group_index in self._column_optimizer.group_types_to_query
-        )
+        if self._using_person_on_events:
+            _fields += [
+                f"{self.EVENT_TABLE_ALIAS}.person_id as person_id"
+            ]  # TODO: handle person_id and aggregation_target together in filters]
```

Comment:
Why do we have to query person_id here, when aggregation_target exists? 🤔

Reply:
On line 164 ee/clickhouse/models/property.py in parse_prop_clauses, we have a very not-dumb function that assumes "person_id" is always the match variable. This was working previously because we passed in "f"{self.DISTINCT_ID_TABLE_ALIAS}.person_id" as the argument to search cohorts IN but without the distinct_id join this doesn't apply and we basically need to explicitly pass aggregation_target or always have a person_id field available. I just made changes to `_get_prop_groups` for the alternative solution

Reply:
Sounds like we're good after your fix, and don't need person_id anymore?

Reply:
Although this raises this question for me: Do we have any validation for when there's a cohort filter + groups aggregation? 🤔 . This `person_id` should always be available on the filter clause, shouldn't it

Reply:
Great question: it ended up being a bug that currently exists in prod. If you add a cohort filter to an entity while aggregating by groups on funnels, nothing will return

Reply:
Addressed now so it's a bug fix too

Reply:
lovely!